server/src/test/java/org/eclipse/lsp/cobol/usecases/TestGlobalVariables.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.usecases;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.usecases.engine.UseCaseEngine;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.jupiter.api.Test;
+
+import static org.eclipse.lsp.cobol.service.delegates.validations.SourceInfoLevels.ERROR;
+
+/**
+ * Test Global variables errors.
+ */
+class TestGlobalVariables {
+  private static final String GLOBAL_USAGE =
+      "       Identification Division.\n"
+          + "       Program-Id. 'P1'.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01 {$*FOO} pointer GLOBAL.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO}.\n"
+          + "       Identification Division.\n"
+          + "       Program-Id. 'P11'.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO}.\n"
+          + "       End Program 'P11'.\n"
+          + "       End Program 'P1'.\n"
+          + "       Identification Division.\n"
+          + "       Program-Id. 'P2'.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO|1}.\n"
+          + "       End Program 'P2'.";
+
+  private static final String VARIABLE_SHADOWING =
+      "       Identification Division.\n"
+          + "       Program-Id. 'P1'.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01 {$*FOO} GLOBAL.\n"
+          + "        05 {$*BAR} POINTER.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$BAR}.\n"
+          + "           DISPLAY FOO.\n"
```

Comment:
Why isn't it marked neither as a variable nor as an error?

Reply:
Well, because error messages come from syntax tree and `FOO` usage is not an error. But references and definitions collected in CobolVisitor the old way. That means it looks like one context where `FOO` defined twice. This generates an error (but we ignore it) and doesn't set up usage links.